python/hawkey/subject-py.cpp

```diff
@@ -276,40 +278,47 @@ get_best_parser(_SubjectObject *self, PyObject *args, PyObject *kwds, HyNevra *n
     PyObject *with_provides = NULL;
     PyObject *with_filenames = NULL;
     PyObject *with_src = NULL;
+    PyObject *exclude_flags_obj = NULL;
     const char *kwlist[] = {"sack", "with_nevra", "with_provides", "with_filenames", "forms",
-        "with_src", NULL};
+        "with_src", "exclude_flags", NULL};
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!|O!O!O!OO!", (char**) kwlist, &sack_Type, &sack,
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!|O!O!O!OO!O", (char**) kwlist, &sack_Type, &sack,
         &PyBool_Type, &with_nevra, &PyBool_Type, &with_provides,
-        &PyBool_Type, &with_filenames, &forms, &PyBool_Type, &with_src)) {
-        return NULL;
+        &PyBool_Type, &with_filenames, &forms, &PyBool_Type, &with_src, &exclude_flags_obj)) {
+        return std::make_pair(nullptr, std::unique_ptr<libdnf::Nevra>());
     }
     std::vector<HyForm> cforms;
     if ((forms != NULL) && (forms != Py_None) &&
         ((!PyList_Check(forms)) || (PyList_Size(forms) > 0))) {
         cforms = fill_form<HyForm, _HY_FORM_STOP_>(forms);
         if (cforms.empty())
-            return NULL;
+            return std::make_pair(nullptr, std::unique_ptr<libdnf::Nevra>());
```

Comment:
fill_form sets an error when cforms.empty(). To propagate it it must return NULL.